### PR TITLE
[Batch] azure-batch 10.0.0 / API v2020-09-01.12.0

### PR DESCRIPTION
**Pull request by gingi**
* Batch data plane (azure-batch) upgraded to 10.0.0
* `az batch job task-counts` now returns a complex object including both task counts as well as task slot counts
* Updated Batch testing README

**Description**
Upgrades the Batch data plane to 10.0.0 (API version 2020-09-01.12.0) for task slot support

**Testing Guide**
Testing the Batch module is described in the [README file](https://github.com/Azure/azure-cli/blob/feature/batch-data-plane-10.0.0/src/azure-cli/azure/cli/command_modules/batch/tests/latest/README.md).

**History Notes**
[Batch] Upgrades data plane to [azure batch 10.0.0](https://pypi.org/project/azure-batch/10.0.0/)
[Batch] BREAKING CHANGE: az batch job task-counts: Change the output from a JSON object returning task counts to a complex JSON object that includes task counts (`taskCounts`) as well as task slot counts (`taskSlotCounts`).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Batch

**Review by dpwatrous — APPROVED** [review]

**Comment by dpwatrous** [comment]
@bgklein since we hadn't updated the code owners file before this PR, you'll have to approve this one.

**Review by bgklein — APPROVED** [review]

**Comment by bgklein** [comment]
@dpwatrous we don't have permissions to merge anyways but lgtm

**Review by qwordy — APPROVED** [review]

**Pull request merged**
